File: client/Client/Util/Serializer.cs
using Newtonsoft.Json;
using System;

namespace LoadRunnerClient
{
	/// <summary>
	/// Serializes and Deserializes Objects to and from a JSON-String
	/// Uses <see cref="JsonConvert"/>
	/// </summary>
    public class Serializer
    {
		/// <summary>
		/// Serializes object to JSON-String
		/// </summary>
		/// <param name="obj">object to be serialized</param>
		/// <returns>serialized <paramref name="obj"/></returns>
        public static string Serialize(Object obj)

        {
            return JsonConvert.SerializeObject(obj);
        }

		/// <summary>
		/// Deserializes string to given Object <typeparamref name="T"/>
		/// </summary>
		/// <typeparam name="T">Type of object that is deserialized</typeparam>
		/// <param name="str">incoming string to be deserialized</param>
		/// <returns>deserialized object of type <typeparamref name="T"/></returns>
        public static T Deserialize<T>(string str)
        {
            return JsonConvert.DeserializeObject<T>(str);
        }
    }
}

File: server/src/main/java/de/hsrm/mi/swtpro03/server/utils/Serializer.java
package de.hsrm.mi.swtpro03.server.utils;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.ObjectReader;
import com.fasterxml.jackson.databind.ObjectWriter;

import java.io.IOException;

/**
 * Serializes and Deserializes Objects to and from a JSON-String
 * @author tschn002
 *
 */
public class Serializer {
	
	private static ObjectMapper jsonMapper = new ObjectMapper();

	/**
	 * Deserializes string to given Object-Type
	 * @param str Incoming string to be deserialized
	 * @param type Type of object that is deserialized
	 * @return The deserialized object
	 */
	public static Object deserialize(String str, Class<?> type){
		Object deserialized = null;
		try {
			ObjectReader objectReader = jsonMapper.readerFor(type);
			deserialized = objectReader.readValue(str);
		} catch (IOException e) {
			System.err.println("[Serializer] Failed to deserialize object of type \"" + type.getName() + "\"");
			e.printStackTrace();
		}
		return deserialized;
	}

	/**
	 * Serializes object to JSON-String
	 * @param object Object to be serialized
	 * @param type The type of Object to be serialized
	 * @return The object serialized to a string
	 */
	public static String serialize(Object object,Class<?> type) {
		String serialized = null;
		try {
			ObjectWriter objectWriter = jsonMapper.writerFor(type);
			serialized = objectWriter.writeValueAsString(object);
		} catch (JsonProcessingException e) {
			System.err.println("[Serializer] Failed to serialize object of type \"" + type.getName() + "\"");
			e.printStackTrace();
		}
		return serialized;
	}
}
